backend/controllers/badge.controller.js

```diff
@@ -0,0 +1,18 @@
+const { Badge } = require('../models')
+
+exports.purchaseBadge = async (req, res) => {
+    const { badgeName } = req.body
+    const user = req.user
+    try {
+        let badge = await Badge.find({ name: badgeName })
+        user.badges.push(badge)
+        user.points -= badge.points
+
+        await user.save()
+        await badge.save()
```

Comment:
not required, you are not making any change to badge, only change in user is made

Reply:
Okay Got it